src/sortStudents.ts

```diff
@@ -1,7 +1,96 @@
-// describe Student type
-// create and export SortType enum
-// create SortOrder type
+interface Student {
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
+}
+
+export enum SortType {
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grade',
+}
+
+type SortOrder = 'asc' | 'desc';
+
+function sumGrades(student: Student) :number {
+  return student.grades.reduce(
+    (sum: number, n: number) => sum + n, 0,
+  ) / student.grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const studentsCopy = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+      if (order === 'asc') {
```

Comment:
```suggestion
      if (order === 'asc') {
```
Use enum here instead of string